lib/model.js

```diff
@@ -19,6 +19,14 @@ const Hooks = require('./hooks');
 const associationsMixin = require('./associations/mixin');
 const Op = require('./operators');
 
+// This list will quickly become dated, but failing to maintain this list just means
+// we won't throw a warning when we should. At least most common cases will forever be covered
+// so we stop throwing erroneous warnings when we shouldn't.
+const validQueryKeywords = new Set(['where', 'attributes', 'paranoid', 'include', 'order', 'limit', 'offset',
```

Comment:
This optimization is acceptable, we should not be creating this every time we check for invalid options